src/shogun/features/CombinedFeatures.h

```diff
@@ -185,7 +185,7 @@ class CCombinedFeatures : public CFeatures
 
 	protected:
 		/** feature array */
-		CDynamicObjectArray* feature_array;
+		std::vector<CFeatures*> feature_array;
```

Comment:
ah I see. nice!

Reply:
ah wait. Aren't there reference counting issues with this? CDynamicObjectArray increases/decreases ref counters of the objects inside, which isnt happening here anymore as far as I can see?

Reply:
Yes, there are issues with this, but I think once this is all smart pointer it'll be fine. I was getting issues before so I changed this a bit. I'll rebase when smart pointers are merged, pretty sure @vigsterkr changed all CDynamicObjectArray with std::vector

Reply:
ah ok then it will be `std::vector<shared_ptr<CFeautres>>`?

Reply:
Yup, I just didn't mess with that because then I would have to rewrite more things, and this just made it work!

Reply:
Sure thing! We can tune these things in a second round